pallets/msa/src/lib.rs

```diff
@@ -1655,6 +1643,16 @@ impl<T: Config + Send + Sync> CheckFreeExtrinsicUse<T> {
 			key_count == 1,
 			InvalidTransaction::Custom(ValidityError::InvalidMoreThanOneKeyExists as u8)
 		);
+
+		let has_delegations: bool =
+			DelegatorAndProviderToDelegation::<T>::iter_key_prefix(DelegatorId(msa_id))
```

Comment:
ohh are you sure this does not invoke a read per item, which is probably not being caught by the benchmarks?